### PR TITLE
[Test] Add unit tests for srt/mem_cache/hicache_storage.py

### PR DESCRIPTION
**Pull request by varvelworld**
## Motivation

This PR adds targeted CPU-only unit coverage for `srt/mem_cache/hicache_storage.py`.
The goal is to lock down HiCache storage behavior that is easy to regress but not always obvious from the implementation alone, especially:
- `HiCacheFile` key / path derivation
- MLA vs non-MLA `config_suffix` behavior
- `batch_exists_v2` prefix-hit semantics across KV and extra pools
- v1 / v2 batch I/O helpers

## Modifications

This PR adds a new unit test file:
- `test/registered/unit/mem_cache/test_hicache_storage_unit.py`
The new tests cover:
- `PoolTransferResult` update semantics
- `HiCacheFile.config_suffix` construction
- `_get_suffixed_key`, `_get_component_key`, and `_get_component_path`
- basic file-backed I/O: `set`, `get`, `exists`, `clear`
- v1 batch wrappers: `batch_get`, `batch_set`
- `batch_exists_v2` behavior for:
  - KV-only prefix hits
  - `ALL_PAGES`
  - `TRAILING_PAGES`
  - multi-pool min-prefix behavior
  - unrelated-file isolation
- v2 batch I/O pipeline:
  - `batch_get_v2`
  - `batch_set_v2`
  - `_batch_io_v2`
  - host index / page size edge cases
This PR also improves the test-side documentation inside the new file to make the HiCache key/path pipeline more explicit, including:
- what a page hash key represents
- why page hash keys are prefix-chained
- how `config_suffix`, component keys, and on-disk paths relate
- why MLA omits TP fields in the storage suffix
Related to #20865.

## Test Plan
```
python3 -m pytest test/registered/unit/mem_cache/test_hicache_storage_unit.py
================================================= test session starts =================================================
platform darwin -- Python 3.12.10, pytest-9.0.3, pluggy-1.6.0
rootdir: /Users/varvel/code/github/sglang/test
configfile: pytest.ini
plugins: cov-7.1.0, anyio-4.13.0
collected 50 items

test/registered/unit/mem_cache/test_hicache_storage_unit.py ..................................................  [100%]

================================================== warnings summary ===================================================
../../../../../Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/_pytest/config/__init__.py:1434
  /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: asyncio_mode

    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

python/sglang/srt/layers/attention/fla/utils.py:223
  /Users/varvel/code/github/sglang/python/sglang/srt/layers/attention/fla/utils.py:223: UserWarning: Triton is not supported on current platform, roll back to CPU.
    warnings.warn(

../../../../../Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/torch/jit/_script.py:365: 14 warnings
  /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/torch/jit/_script.py:365: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

python/sglang/srt/layers/quantization/awq/awq.py:52
  /Users/varvel/code/github/sglang/python/sglang/srt/layers/quantization/awq/awq.py:52: UserWarning: Only CUDA, HIP and XPU support AWQ currently.
    warnings.warn(f"Only CUDA, HIP and XPU support AWQ currently.")

python/sglang/srt/layers/quantization/gguf.py:63
  /Users/varvel/code/github/sglang/python/sglang/srt/layers/quantization/gguf.py:63: UserWarning: Only CUDA, MUSA and NPU support GGUF quantization currently.
    warnings.warn(f"Only CUDA, MUSA and NPU support GGUF quantization currently.")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================================== 50 passed, 20 warnings in 2.86s ===========================================
```

## Accuracy Tests

N/A. This PR only adds unit tests and test documentation comments. It does not change model forward logic, kernels, or model output behavior.

## Speed Tests and Profiling

N/A. This PR does not change inference code paths or serving performance.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

**Comment by hnyls2002** [comment]
Closing as superseded. Most of the intended coverage has since landed via PR #26670 which merged test/registered/unit/mem_cache/test_hicache_file_lru_unit.py (34 tests) covering config_suffix construction including CP-awareness, _get_suffixed_key, batch_exists_v2 semantics, basic file-backed I/O, and LRU eviction paths. Thanks @varvelworld; remaining gaps (e.g. v1 batch wrappers) can be proposed as a small targeted follow-up against the new test file.
